broker/cloud_run/lsst/ps_to_storage/main.py

```diff
@@ -77,166 +67,54 @@ def run():
+    except PreconditionFailed:
+        # this alert is a duplicate. drop it.
+        return "", HTTP_204
+
+    # publish alerts to appropriate Pub/Sub topics
+    TOPIC_ALERTS.publish(alert)  # deduplicated "alerts" stream
+    json_dict = _reformat_alert_data_to_valid_json(alert)
+    publish_valid_json_stream(
+        topic_name=TOPIC_BIGQUERY_IMPORT.name,
+        message=json_dict,
         attributes={
-            "diaObjectId": str(alert_dict["diaObject"]["diaObjectId"]),
-            "diaSourceId": str(alert_dict["diaSource"]["diaSourceId"]),
-            "schema_version": schema_version,
-            **attributes,
+            "schema_version": alert.schema.version,
         },
     )
 
-    # publish the alert as a JSON message to the bigquery-import topic
-    TOPIC_BIGQUERY_IMPORT.publish(
-        _reformat_alert_data_to_valid_json(
-            alert_dict, attributes={"schema_version": schema_version}
-        )
-    )
-
-
-def deserialize_confluent_wire_header(raw):
-    """Parses the byte prefix for Confluent Wire Format-style Kafka messages.
-    Parameters
-    ----------
-    raw : `bytes`
-        The 5-byte encoded message prefix.
-    Returns
-    -------
-    schema_version : `int`
-        A version number which indicates the Confluent Schema Registry ID
-        number of the Avro schema used to encode the message that follows this
-        header.
-    """
-    _, version = _ConfluentWireFormatHeader.unpack(raw)
-
-    return version
-
-
-def generate_alert_filename(aname: dict) -> str:
-    """
-    Generate the filename of an alert stored to a Cloud Storage bucket.
-
-    Args:
-        aname:
-            Components to create the filename. Required key/value pairs are those needed to create a parsed filename.
-            Extra keys are ignored.
-
-    Returns:
-        str: The formatted filename as "{schema_version}/{YYYY-MM-DD}/{objectId}/{sourceId}.{format}".
-    """
-
-    schema_version = aname["schema_version"]
-    alert_date = aname["alert_date"]
-    object_id = aname["objectId"]
-    source_id = aname["sourceId"]
-    file_format = aname["format"]
-
-    return f"{schema_version}/{alert_date}/{object_id}/{source_id}.{file_format}"
+    return "", HTTP_204
 
 
-def create_file_metadata(alert_dict: dict, event_id: str) -> dict:
+def _create_file_metadata(alert: pittgoogle.Alert, event_id: str) -> dict:
     """Return key/value pairs to be attached to the file as metadata."""
 
     metadata = {"file_origin_message_id": event_id}
-    metadata["diaObjectId"] = alert_dict["diaObject"]["diaObjectId"]
-    metadata["diaSourceId"] = alert_dict["diaSource"]["diaSourceId"]
-    metadata["ra"] = alert_dict["diaSource"]["ra"]
-    metadata["dec"] = alert_dict["diaSource"]["dec"]
+    metadata["diaObjectId"] = alert.objectid
+    metadata["diaSourceId"] = alert.sourceid
+    metadata["ra"] = alert.get("ra")
+    metadata["dec"] = alert.get("dec")
```

Comment:
Hmm, looks like we're mixing survey-specific field names with pittgoogle generic ones. I think we should go with survey-specific names. I've also added ra and dec as `Alert` properties.

```suggestion
    metadata[alert.get_key("objectid")] = alert.objectid
    metadata[alert.get_key("sourceid")] = alert.sourceid
    metadata[alert.get_key("ra")] = alert.ra
    metadata[alert.get_key("dec")] = alert.dec
```